bizflycloud/resource_bizflycloud_network_interface.go

```diff
@@ -43,62 +44,51 @@ func resourceBizflyCloudNetworkInterfaceCreate(d *schema.ResourceData, meta inte
 	}
 	d.SetId(networkInterface.ID)
 
-	actionPayload := &gobizfly.ActionNetworkInterfacePayload{
-		Action:   d.Get("action").(string),
-		ServerID: d.Get("server_id").(string),
-	}
-	if v, ok := d.GetOk("security_groups"); ok {
-		for _, id := range v.([]interface{}) {
-			actionPayload.SecurityGroups = append(actionPayload.SecurityGroups, id.(string))
-		}
-	}
-	_, err = client.NetworkInterface.Action(context.Background(), d.Id(), actionPayload)
-	if err != nil {
-		return fmt.Errorf("Error when action network interface: %v", err)
-	}
-
 	return resourceBizflyCloudNetworkInterfaceRead(d, meta)
 }
 
 func resourceBizflyCloudNetworkInterfaceRead(d *schema.ResourceData, meta interface{}) error {
-	if err := dataSourceBizflyCloudNetworkInterfaceRead(d, meta); err != nil {
-		return err
-	}
-	return nil
-}
-
-func resourceBizflyCloudNetworkInterfaceUpdate(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*CombinedConfig).gobizflyClient()
-	updatePayload := &gobizfly.UpdateNetworkInterfacePayload{
-		Name: d.Get("name").(string),
-	}
-	networkInterface, err := client.NetworkInterface.Update(context.Background(), d.Id(), updatePayload)
+	networkInterface, err := client.NetworkInterface.Get(context.Background(), d.Id())
 	if err != nil {
-		return fmt.Errorf("Error when update network interface: %s, %v", d.Id(), err)
+		return fmt.Errorf("Error read network interface network %s: %w", d.Id(), err)
 	}
 	d.SetId(networkInterface.ID)
+	_ = d.Set("name", networkInterface.Name)
+	_ = d.Set("network_id", networkInterface.NetworkID)
+	_ = d.Set("attached_server", networkInterface.AttachedServer)
+	_ = d.Set("fixed_ip", networkInterface.FixedIps[0].IPAddress)
+	_ = d.Set("status", networkInterface.Status)
+	_ = d.Set("created_at", networkInterface.CreatedAt)
+	_ = d.Set("updated_at", networkInterface.UpdatedAt)
+	if err := d.Set("fixed_ips", readFixedIps(networkInterface.FixedIps)); err != nil {
+		return fmt.Errorf("error setting fixed_ips: %w", err)
+	}
 
-	actionPayload := &gobizfly.ActionNetworkInterfacePayload{
-		Action:   d.Get("action").(string),
-		ServerID: d.Get("server_id").(string),
+	if err := d.Set("security_groups", readSecurityGroups(networkInterface.SecurityGroups)); err != nil {
+		return fmt.Errorf("error setting security_groups: %w", err)
 	}
-	if v, ok := d.GetOk("security_groups"); ok {
-		for _, id := range v.([]interface{}) {
-			actionPayload.SecurityGroups = append(actionPayload.SecurityGroups, id.(string))
+	return nil
+}
+
+func resourceBizflyCloudNetworkInterfaceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*CombinedConfig).gobizflyClient()
+	if d.HasChange("name") {
```

Comment:
how about update sec group and attached server ? Does it support ?

Reply:
No. We've agreed in the meeting that we should have implement attach server, sec group in one place (resource server), otherwise it's so mess, confusing for both system and user side.

Reply:
So do we need remove option security_groups in the schema ?

Reply:
I think removing it is not necessary. Just let it there for reading operation.